lib/calendarClient.ts

```diff
@@ -113,7 +114,8 @@ const o365Auth = (credential) => {
 const userData = Prisma.validator<Prisma.UserArgs>()({
   select: { name: true, email: true, timeZone: true },
 });
-export type Person = Prisma.UserGetPayload<typeof userData>;
+
+export type Person = Ensure<Prisma.UserGetPayload<typeof userData>, "email" | "name">;
```

Comment:
@KATT  what do you think of this pattern? `Ensure` specifically

Reply:
This is super cool but I'd prefer if we could find a way to write code that ever really needed [advanced] generics as it's quite inaccessible.

I feel like generics doesn't really "belong" in application code itself (and only in libs) and should be avoidable.

Reply:
This one example could be avoidable by making `email` and `name` non-nullable in `schema.prisma`

Reply:
Further thinking

... if we could make everything based on type inference people only needed to learn to do `ReturnType<typeof x>` and everything else would just flow

I don't have a straight answer on what is the best approach

Reply:
Maybe we can 

```suggestion
type Person = {
  name: string;
  email: string;
  // ...
}
```


?

Reply:
@KATT Those are valid points, and yes. If we can make `email` and `name` non-nullable in the schema, that would solve a lot of typing headaches, since we need those values in many places.

Reply:
Yeah - let's do that! I know `username`  + `email` should be non-nullable - unsure about `name` (but maybe we can just prefill that as `username`?)

Reply:
Yeah let's followup on slack.